### PR TITLE
Add extension points to tests.yml template

### PR DESCRIPTION
**Pull request by weshaggard**
Add extensions that allow env varaibles to be passed to the template
as well as enables additional presteps to be run at the begining of
the test job.

PTAL @danieljurek @mitchdenny 

cc @azure/azure-sdk-eng @tg-msft

### CODE REVIEW

File: eng/pipelines/templates/jobs/tests.yml [hunk 1]
```diff
@@ -44,27 +51,31 @@ jobs:
           scriptPath: '$(Build.BinariesDirectory)/sdk-tools/scripts/python/verify_agent_os.py'
           arguments: $(OSName)
 
-      - task: DotNetCoreInstaller@0
-        displayName: 'Use .NET Core sdk $(DotNetCoreVersion)'
+      - task: DotNetCoreInstaller@2
```
Comment (weshaggard) [inline eng/pipelines/templates/jobs/tests.yml]: @danieljurek looks like all the .NET Test pipelines are failing after I changed the dotnet variables the other day. This should fix those.

**Review by danieljurek — APPROVED** [review]
LGTM

**Comment by weshaggard** [comment]
@tg-msft it doesn't look like the storage live tests are running yet as there is a bunch of Skipping still going on but I"m going to move forward to merge these yml changes and then we can iterate just on the storage tests in another PR.

**Pull request merged**